crates/core/src/config.rs

```diff
@@ -191,7 +241,18 @@ impl ConfigArgs {
             },
             log_level: self.log_level.unwrap_or(tracing::log::LevelFilter::Info),
             config_paths: self.config_paths.build()?,
-        })
+        };
+
+        if should_persist {
+            let mut file = File::create(this.config_paths.data_dir.join("config.toml"))?;
```

Comment:
I am not sure config should go into `data_dir` there is probably a directory where configuration should live in each different platform which is not the same as data_dir, take a look at this.